baseline/utils/final_evaluation.py

```diff
@@ -0,0 +1,189 @@
+        normalize_audio=False, n_samples=-1)
+    return data_loader.get_generator(batch_size=1, num_workers=1)
+
+
+def load_sudo_rm_rf_model(path):
+    model = improved_sudormrf.SuDORMRF(
+        out_channels=256,
+        in_channels=512,
+        num_blocks=8,
+        upsampling_depth=7,
+        enc_kernel_size=81,
+        enc_num_basis=512,
+        num_sources=2,
+    )
+    # You can load the state_dict as here:
+    model.load_state_dict(torch.load(path))
+    print(f"Fetched model from: {path}")
+    return model
+
+
+if __name__ == "__main__":
+    fs = 16000
+    meter = pyln.Meter(fs)
+    args = get_args()
+    hparams = vars(args)
+    if hparams['dataset'] == 'chime':
+        test_generator = get_chime_generator()
+        flag = False
+    elif hparams['dataset'] == 'librichime':
+        test_generator = get_librichime_generator()
+        flag = True
+    else:
+        ValueError('Unknown dataset. It should be either chime or librichime.')
+        
+    if not hparams['evaluate_only_input_mixture']:
+        model = load_sudo_rm_rf_model(hparams['model_checkpoint'])
+        model = model.cuda()
+        model.eval()
+    else:
+        model = None
+    test_tqdm_gen = tqdm(enumerate(test_generator), desc='Eval on 16kHz ' + hparams['dataset'] + ' 1 speaker')
+    res_dic = {
+            "sig_mos": [],
+            "bak_mos": [],
+            "ovr_mos": [],
+    }
+    gen_len = len(test_generator)
+    with torch.no_grad():
+        for j, input_argument in test_tqdm_gen:
+            if flag:
+                speakers, noise = input_argument
+                gt_speaker_mix = speakers.sum(1, keepdims=True) 
+                mixture = noise + gt_speaker_mix
+            else:
+                mixture = input_argument
+                    
+            if hparams['evaluate_only_input_mixture']:
+                s_est_speech = mixture[0].cpu().numpy().squeeze()
+            else:
+                file_length = mixture.shape[-1]
+                min_k = int(np.ceil(np.log2(file_length/16000)))
+                padded_length = 2**max(min_k, 1) * 16000
+
+                input_mix = torch.zeros((1, padded_length), dtype=mixture.dtype)
+                input_mix[..., :file_length] = mixture
+
+                input_mix = input_mix.unsqueeze(1).cuda()
+                input_mix_std = input_mix.std(-1, keepdim=True)
+                input_mix_mean = input_mix.mean(-1, keepdim=True)
+                input_mix = (input_mix - input_mix_mean) / (input_mix_std + 1e-9)
+
+                student_estimates = model(input_mix)
+                student_estimates = mixture_consistency.apply(student_estimates, input_mix)
+
+                s_est_speech = student_estimates[0, 0, :file_length].detach().cpu().numpy().squeeze()
+
+            # Peak normalization to 0.7
+            s_est_speech -= s_est_speech.mean()
+            s_est_speech /= np.abs(s_est_speech).max() + 1e-9
+            s_est_speech *= 0.7
```

Comment:
Why this value?

Reply:
At least comment here, and explain why this value, also make sure the same normalization is applied during training.

Reply:
This is just to make sure that clipping does not happen after loudness normalization. I added a comment, and modified the training script as well.

Reply:
Please write the reason that you chose to use 0.7 for peak normalization.